src/components/Header/Header.tsx

```diff
@@ -0,0 +1,175 @@
+import React, {ReactNode} from 'react'
+import {StyleSheet, TouchableOpacity} from 'react-native'
+import type {ImageStyle, StyleProp, TextStyle, ViewStyle} from 'react-native'
+import styled from 'styled-components/native'
+import {TextBold} from '../Text/Text'
+import {Images} from '../../theme'
+import {hitSlop, metrics} from '../../helpers'
+import {useTheme} from '../../hooks'
+
+export interface HeaderProps {
+  /**
+   * The top padding for the notch of device, usually we will use react-native-safe-area-context to get the inset
+   */
+  paddingTop?: number
+  /**
+   * Height of the header
+   */
+  height?: number
+  /**
+   * Style of Header container
+   */
+  containerStyle?: StyleProp<ViewStyle>
+  /**
+   * Width of Header bottom border
+   */
+  borderBottomWidth?: number
+  /**
+   * Color of Header bottom border
+   */
+  borderBottomColor?: string
+  /**
+   * Title of the header, will be use if no custom CenterComponent is defined
+   */
+  title?: string
+  /**
+   * The color of Header title
+   */
+  titleColor?: string
+  /**
+   * The custom style of Header title
+   */
+  titleStyle?: StyleProp<TextStyle>
+  /**
+   * Background color of the Header
+   */
+  backgroundColor?: string
+  /**
+   * Show the back to previous screen icon, default value is true
+   */
+  hasBackButton?: boolean
+  /**
+   * The color of back to previous screen icon
+   */
+  backIconColor?: string
+  /**
+   * The action for the back to previous screen icon
+   */
+  backButtonPress?: () => void
+  /**
+   * The left component of the Header
+   */
+  leftComponent?: ReactNode
+  /**
+   * The style of left component of the Header
+   */
+  leftStyle?: StyleProp<ViewStyle>
+  /**
+   * The center component of the Header, will replace Title
+   */
+  centerComponent?: ReactNode
+  /**
+   * The style of center component of the Header
+   */
+  centerStyle?: StyleProp<ViewStyle>
+  /**
+   * The right component of the Header
+   */
+  rightComponent?: ReactNode
+  /**
+   * The style of right component of the Header
+   */
+  rightStyle?: StyleProp<ViewStyle>
```

Comment:
Do we need the comment above every props? I think the variable name is quite clear to understand.

Reply:
I commented to keep the convention with other previous components, which we also wrote comments.